packages/calcite-components/src/utils/interactive.tsx

```diff
@@ -111,7 +111,7 @@ function removeInteractionListeners(element: HTMLElement): void {
   );
 }
 
-export interface InteractiveContainerProps extends JSXAttributes {
+export interface InteractiveContainerProps extends JSXAttributes<HTMLDivElement> {
```

Comment:
This could stay the same as we're not using any props from a particular HTML element.

Applies to `XButtonProps` too.

Reply:
Without the change in XButtonProps, I was getting a type error here:

https://github.com/Esri/calcite-design-system/blob/0b03a578db450106360b074dad985a40582a6220/packages/calcite-components/src/components/functional/XButton.tsx#L31

The issue is that in Lumina JSX, the ref prop can be assigned either `(element: HTMLButtonElement) => void` or `{ value: HTMLButtonElement }` (like a RefObject type in React).

The RefObject type is causing issues.
By default `JSXAttributes` interface is assuming `HTMLElement` type.
That means the ref prop type in the `<XButton>` is `RefObject<HTMLElement>`, which is not assignable to the `RefObject<HTMLButtonElement>` in the `<button>`.

The same issue is present in React since they also have object refs. Not an issue in Stencil since they don't have object refs, only callback refs.

---

The change in `<InteractiveContainer>` is not necessary, but I done it for consistency, and to avoid issues if someone does add a ref prop to it in the future.

Reply:
I could be misunderstanding, but it seems this change is conflating the prop types with the ref type. These can be different depending on the implementation, right? 

It doesn't seem like we're using `XButton`'s `ref` property, so we could alternatively drop it for the time being.

Reply:
> I could be misunderstanding, but it seems this change is conflating the prop types with the ref type

`JSXAttributes` represents the common props type, including the `ref` prop.
The generic parameter you pass to `JSXAttributes` (eg `JSXAttributes<HTMLDivElement>`) dictates the type used in the `ref`

> It doesn't seem like we're using XButton's ref property, so we could alternatively drop it for the time being.

Sounds good, I will go ahead and remove it.

Reply:
You're right. I was thinking about other use cases where we might want to provide a `ref` to another part of the functional component, but this would probably be better off as a separate property in the props interface.